Dashboard/app/js/lib/views/reports/export-reports-views.js

```diff
@@ -1,140 +1,28 @@
 /*global Ember, $, FLOW */
 
 FLOW.ReportLoader = Ember.Object.create({
-  criteria: null,
-  timeout: 30000,
-  requestInterval: 3000,
-
-  payloads: {
-	DATA_CLEANING: {
-	  surveyId: '75201',
-	  exportType: 'DATA_CLEANING',
-	  opts: {
-		exportMode: 'DATA_CLEANING',
-		lastCollection: 'false',
-	  }
-	},
-	DATA_ANALYSIS: {
-	  surveyId: '75201',
-	  exportType: 'DATA_ANALYSIS',
-	  opts: {
-		exportMode: 'DATA_ANALYSIS',
-	  }
-	},
-    COMPREHENSIVE: {
-      surveyId: '75201',
-      exportType: 'COMPREHENSIVE',
-      opts: {
-        exportMode: 'COMPREHENSIVE',
-      }
-    },
-    GEOSHAPE: {
-      surveyId: '75201',
-      exportType: 'GEOSHAPE',
-      opts: {
-        questionId: '12345'
-      }
-    },
-    SURVEY_FORM: {
-      surveyId: '75201',
-      exportType: 'SURVEY_FORM',
-      opts: {}
-    }
-  },
-
   load: function (exportType, surveyId, opts) {
-    var criteria;
-
-    if (this.get('criteria')) {
-      return;
-    }
+    FLOW.selectedControl.set('selectedReportExport', FLOW.store.createRecord(FLOW.Report, {}));
+    var newReport = FLOW.selectedControl.get('selectedReportExport');
 
     Ember.assert('exportType param is required', exportType !== undefined);
     Ember.assert('surveyId param is required', surveyId !== undefined);
 
-    criteria = Ember.copy(this.get('payloads')[exportType]);
-    criteria.surveyId = '' + surveyId;
-    criteria.baseURL = location.protocol + '//' + location.host;
-
-    criteria.opts.imgPrefix = FLOW.Env.photo_url_root;
-    criteria.opts.uploadUrl = FLOW.Env.surveyuploadurl;
-    criteria.opts.appId = FLOW.Env.appId;
-
     if (opts) {
       Ember.keys(opts).forEach(function (k) {
-        criteria.opts[k] = opts[k];
+        newReport.set(k, opts[k])
       });
     }
 
-    criteria.opts.lastCollection = '' + (exportType === 'DATA_CLEANING' && FLOW.selectedControl.get('selectedSurveyGroup').get('monitoringGroup') && !!FLOW.editControl.lastCollection);
-    criteria.opts.email = FLOW.currentUser.email;
-    criteria.opts.flowServices = FLOW.Env.flowServices;
-
-    this.set('criteria', criteria);
-    FLOW.savingMessageControl.numLoadingChange(1);
-    this.requestReport();
-  },
-
-  requestReport: function () {
-	this.set('processing', true);
-	$.ajax({
-	  url: FLOW.Env.flowServices + '/generate',
-	  data: {
-		criteria: JSON.stringify(this.get('criteria'))
-	  },
-	  jsonpCallback: 'FLOW.ReportLoader.handleResponse',
-	  dataType: 'jsonp',
-	  timeout: this.timeout
-	});
+    newReport.set('reportType', exportType);
+    newReport.set('formId', surveyId);
+    newReport.set('filename', '');
+    newReport.set('state', 'QUEUED');
+    //newReport.set('lastCollectionOnly', ('' + (exportType === 'DATA_CLEANING' && FLOW.selectedControl.get('selectedSurveyGroup').get('monitoringGroup') && !!FLOW.editControl.lastCollection));
```

Comment:
Should we remove this?

Reply:
left it there as a placeholder for after we merge #2585